packages/sdk/shell/src/components/DeviceList/DeviceList.tsx

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { CaretRight, Database, Plus, Power } from '@phosphor-icons/react';
+import React from 'react';
+
+import { type Device } from '@dxos/react-client/halo';
+import { Avatar, Button, List, useTranslation, Tooltip } from '@dxos/react-ui';
+import { descriptionText, getSize, mx } from '@dxos/react-ui-theme';
+
+import { DeviceListItem } from './DeviceListItem';
+import { type DeviceListProps, type AgentFormProps } from './DeviceListProps';
+
+export const DeviceList = ({
+  devices,
+  onClickAdd,
+  onClickEdit,
+  onClickReset,
+  onClickJoinExisting,
+  agentStatus,
+  agentActive,
+  agentHostingEnabled,
+  onAgentDestroy,
+  onAgentCreate,
+  validationMessage,
+}: DeviceListProps & AgentFormProps) => {
+  const { t } = useTranslation('os');
+  return (
+    <>
+      <h2 className={mx(descriptionText, 'text-center mbs-4')}>{t('devices heading')}</h2>
+      {devices.length > 0 && (
+        <List>
+          {devices.map((device: Device) => {
+            return (
+              <DeviceListItem
+                key={device.deviceKey.toHex()}
+                device={device}
+                onClickEdit={() => onClickEdit?.(device)}
+                {...{ onClickReset, onClickJoinExisting }}
+              />
+            );
+          })}
+        </List>
+      )}
+      <Button
+        variant='ghost'
+        classNames='justify-start gap-2 !pis-0 !pie-3'
+        data-testid='devices-panel.create-invitation'
+        onClick={onClickAdd}
+      >
+        <div role='img' className={mx(getSize(8), 'm-1 rounded-sm surface-input grid place-items-center')}>
+          <Plus weight='light' className={getSize(6)} />
+        </div>
+        <span className='grow font-medium text-start'>{t('choose add device label')}</span>
+        <CaretRight weight='bold' className={getSize(4)} />
+      </Button>
+      {agentHostingEnabled && (
```

Comment:
I'd tend to think that this should be its own component and not buried inside the `DeviceList` even if they do end up rendered together